admin/tabs/activity/tracking/ActivityDetail.js

```diff
@@ -25,44 +23,48 @@ export const activityDetail = hoistElemFactory(
             style: {width: 600},
             isOpen: model.detailRecord,
             onClose: () => model.closeDetail(),
-            items: renderDetail(model, rec)
+            item: detail()
         });
     }
 );
 
 
-function renderDetail(model, rec) {
-    const user = rec.impersonating? `${rec.username} as ${rec.impersonating}`: rec.username;
-    return [
-        table({
-            className: 'xh-admin-activity-detail',
-            items: [
-                tbody(
-                    tr(th('User:'), td(user)),
-                    tr(th('Message:'), td(rec.msg)),
-                    tr(th('Category:'), td(rec.category)),
-                    tr(th('Device/Browser:'), td(`${rec.device}/${rec.browser}`)),
-                    tr(th('Agent:'), td(rec.userAgent)),
-                    tr(th('Elapsed (ms):'), td(`${rec.elapsed || ''}`)),
-                    tr(th('Date:'), td(fmtDateTime(rec.dateCreated)))
-                )
-            ]
-        }),
-        jsonInput({
-            omit: !rec.data,
-            value: rec.data,
-            disabled: true,
-            height: 100,
-            width: '100%',
-            editorProps: {lineWrapping: true}
-        }),
-        toolbar(
-            filler(),
-            button({
-                text: 'Close',
-                intent: 'primary',
-                onClick: () => model.closeDetail()
-            })
-        )
-    ];
-}
\ No newline at end of file
+const detail = hoistCmpFactory(
+    ({model}) => {
+        const rec = model.detailRecord,
+            user = rec.impersonating ? `${rec.username} as ${rec.impersonating}` : rec.username;
+
+        return [
```

Comment:
I found that I was getting react console warnings here re. the need for stable keys when returning the collection directly. Believe we want to wrap in a fragment or a div.

Reply:
Good catch -- had just been reading more about fragments yesterday and was reminded of that benefit w.r.t to key warnings.  Will be sure to respect their power more going forward!